sdk/core/Azure.Core/src/Azure.Core.csproj

```diff
@@ -8,11 +8,12 @@
     <PackageTags>Microsoft Azure Client Pipeline</PackageTags>
     <Nullable>enable</Nullable>
     <DefineConstants>$(DefineConstants);AZURE_NULLABLE;HAS_INTERNALS_VISIBLE_CORE</DefineConstants>
-    <TargetFrameworks>$(RequiredTargetFrameworks);net461;net472;net6.0</TargetFrameworks>
+    <TargetFrameworks>$(RequiredTargetFrameworks);net462;net472;net6.0</TargetFrameworks>
```

Comment:
This needs to be updated because Microsoft.Bcl.AsyncInterfaces is not compatible with net461

Reply:
Why? Microsoft.Extensions.Logging says it is compatible with net461: https://apisof.net/catalog/1587d1bdfa66eaa6a3d734408681d41b?fx=net461

Reply:
But this change is fine. I talked to Jesse and he said we should be totally fine to upgrade this target. The only potentially problematic change here is System.Diagnostics.DiagnosticSource.

Reply:
How can use Microsoft.Extensions.Logging APIs on `netstandard2.0`?

Reply:
(And I think we should keep the `net461` TFM as a way to maintain test validation regarding whether we correctly support `netstandard2.0`)

Reply:
> (And I think we should keep the net461 TFM as a way to maintain test validation regarding whether we correctly support netstandard2.0)

Why is net461 needed for that? 461 has been out of support for 2.5 years. Is 462 not as accurate of a representation? That's what the test matrix uses

Reply:
Why does your sample work on `net462` and not `net461`?

Reply:
Because net461 is 2.5 years out of support
```
C:\Users\mredding\.nuget\packages\microsoft.bcl.asyncinterfaces\8.0.0\buildTransitive\net461\Microsoft.Bcl.AsyncInterfa
ces.targets(4,5): warning : Microsoft.Bcl.AsyncInterfaces 8.0.0 doesn't support net461 and has not been tested with it.
 Consider upgrading your TargetFramework to net462 or later. You may also set <SuppressTfmSupportBuildWarnings>true</Su
ppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported config
uration at your own risk. [C:\Users\mredding\source\repos\azure-sdk-for-net\sdk\core\Azure.Core\src\Azure.Core.csproj::
TargetFramework=net461]
```

Reply:
Also, apologies this is a warning not an error, so the build does technically succeed, just with a ton of warnings.

Reply:
Per offline conversation:
 
On net461, a special package is needed to support Microsoft.Extensions.Logging on netstandard2.0, see: https://apisof.net/catalog/1587d1bdfa66eaa6a3d734408681d41b?fx=net461
 
On net462, there is a package that targets net462, see: 
https://apisof.net/catalog/1587d1bdfa66eaa6a3d734408681d41b?fx=net462

Per https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0, it looks like we should still be able to test that we support netstandard2.0 by moving to net462, so I am actually fine with that.